finagle-http/src/main/scala/com/twitter/finagle/http/filter/StatsFilter.scala

```diff
@@ -23,6 +23,12 @@ class StatsFilter[REQUEST <: Request](stats: StatsReceiver)
   private[this] val timeReceiver = stats.scope("time")
   private[this] val responseSizeStat = stats.stat("response_size")
 
+  private[this] val counterCache: Map[String, Counter] =
+    Map.empty.withDefault(statusReceiver.counter(_))
```

Comment:
Somehow I looked at the MetricsStatsReceiver instead of the MemoizedStatsReciever. I cannot find a MemoizedStatsReceiver in Finagle. I do find a Memoized in twitter-util, which is used by Finatra. Should we copy that solution to Finagle?